driver/razerchromacommon.c

```diff
@@ -469,8 +469,16 @@ struct razer_report razer_chroma_standard_matrix_effect_custom_frame(unsigned ch
  */
 struct razer_report razer_chroma_standard_matrix_set_custom_frame(unsigned char row_index, unsigned char start_col, unsigned char stop_col, unsigned char *rgb_data)
 {
+    size_t start_arg_offset = 4;
+    struct razer_report report = {0};
     size_t row_length = (size_t) (((stop_col + 1) - start_col) * 3);
-    struct razer_report report = get_razer_report(0x03, 0x0B, 0x46); // In theory should be able to leave data size at max as we have start/stop
+
+    if (sizeof(struct razer_report) - start_arg_offset < row_length) {
+        printk(KERN_ALERT "razerchroma: RGB data too long\n");
+        return report;
+    }
```

Comment:
Took me a bit to understand why you introduced the `uninit_report` variables but I get it now.

I'd suggest though for simplicity to adjust the `row_length` variable if it's too high (and keep the error print) so we can avoid this dance with having empty reports returned. I think that should also work, no?
Unfortunately we don't have a pointer where we can just return `NULL` and easily check for `NULL` in the caller :wink:

Reply:
Sure thing, both ways should work (in the manner of the OOB memcpy), I thought returning an empty report would be cleaner (since we can't return a NULL ptr).
I will change it tho.